README-quick-start.md

```diff
@@ -0,0 +1,605 @@
+        [6., 7., 6., 9., 0., 5., 4., 0., 6., 8., 5., 2.],
+        [8., 5., 5., 7., 1., 8., 6., 7., 1., 8., 1., 0.]];
+    
+    let (s1, s2) = a.view().split_at(Axis(0), 1);
+    println!("Split a from Axis(0), at index 1:");
+    println!("s1  = \n{}", s1);
+    println!("s2  = \n{}\n", s2);
+    
+    
+    let (s1, s2) = a.view().split_at(Axis(1), 4);
+    println!("Split a from Axis(1), at index 4:");
+    println!("s1  = \n{}", s1);
+    println!("s2  = \n{}\n", s2);
+}
+```
+The output is:
+```
+Split a from Axis(0), at index 1:
+s1  = 
+[[6, 7, 6, 9, 0, 5, 4, 0, 6, 8, 5, 2]]
+s2  = 
+[[8, 5, 5, 7, 1, 8, 6, 7, 1, 8, 1, 0]]
+
+Split a from Axis(1), at index 4:
+s1  = 
+[[6, 7, 6, 9],
+ [8, 5, 5, 7]]
+s2  = 
+[[0, 5, 4, 0, 6, 8, 5, 2],
+ [1, 8, 6, 7, 1, 8, 1, 0]]
+
+```
+
+## Copies and Views
+### View, Ref or Shallow Copy
+As in Rust we have owner ship, so we cannot simply 
+update an element of an array while we have a 
+shared view of it. This will help us write more
+robust code than numpy.
+
+```rust
+use ndarray::prelude::*;
+use ndarray::{Array, Axis};
+
+fn main() {
+
+    let mut a = Array::range(0., 12., 1.).into_shape([3 ,4]).unwrap();
+    println!("a = \n{}\n", a);
+    
+    {
+        let (s1, s2) = a.view().split_at(Axis(1), 2);
+        
+        // with s as a view sharing the ref of a, we cannot update a here
+        // a.slice_mut(s![1, 1]).fill(1234.);
+        
+        println!("Split a from Axis(0), at index 1:");
+        println!("s1  = \n{}", s1);
+        println!("s2  = \n{}\n", s2);
+    }
+    
+    // now we can update a again here, as views of s1, s2 are dropped already
+    a.slice_mut(s![1, 1]).fill(1234.);
+    
+    let (s1, s2) = a.view().split_at(Axis(1), 2);
+    println!("Split a from Axis(0), at index 1:");
+    println!("s1  = \n{}", s1);
+    println!("s2  = \n{}\n", s2);
+}
+```
+The output is:
+```
+a = 
+[[0, 1, 2, 3],
+ [4, 5, 6, 7],
+ [8, 9, 10, 11]]
+
+Split a from Axis(0), at index 1:
+s1  = 
+[[0, 1],
+ [4, 5],
+ [8, 9]]
+s2  = 
+[[2, 3],
+ [6, 7],
+ [10, 11]]
+
+Split a from Axis(0), at index 1:
+s1  = 
+[[0, 1],
+ [4, 1234],
+ [8, 9]]
+s2  = 
+[[2, 3],
+ [6, 7],
+ [10, 11]]
+```
+
+### Deep Copy
+As the usual way in Rust, a `clone()` call will
+make a copy of your array:
+```rust
+use ndarray::prelude::*;
+use ndarray::Array;
+
+fn main() {
+
+    let mut a = Array::range(0., 4., 1.).into_shape([2 ,2]).unwrap();
+    let b = a.clone();
+    
+    println!("a = \n{}\n", a);
+    println!("b clone of a = \n{}\n", a);
+    
+    a.slice_mut(s![1, 1]).fill(1234.);
+    
+    println!("a updated...");
+    println!("a = \n{}\n", a);
+    println!("b clone of a = \n{}\n", b);
+}
+```
+
+The output is:
+```
+a = 
+[[0, 1],
+ [2, 3]]
+
+b clone of a = 
+[[0, 1],
+ [2, 3]]
+
+a updated...
+a = 
+[[0, 1],
+ [2, 1234]]
+
+b clone of a = 
+[[0, 1],
+ [2, 3]]
+```
+
+Noticing that `clone()` will perform a deep copy of the array elements only if you are cloning an owned array (i.e. `Array`).
```

Comment:
Deep copy is not strictly true. When using `clone` it's *always the shallowest copy that is allowed by ownership rules*. That means that an array of `Rc<T>` elements, will `.clone()` each element (i.e increase its refcount, not "deep copy")

Reply:
You could explain that .clone() also copies the array's elements